Add dtype checks in meta dispatch for various ordering ops

matthewhagraphcore opened the pull request:
This adds data type checks for the unsupported bool and complex types for argmax/min topk, sort, minimum, maximum. As listed here:

https://github.com/pytorch/pytorch/blob/0a99b026d6bd0f67dc2c0a20fe3228ddc4144854/torch/testing/_internal/common_methods_invocations.py#L21076

Currently the ops will fail on CPU or CUDA calculation, rather than at meta dispatch stage as with for example max: https://github.com/pytorch/pytorch/blob/0a99b026d6bd0f67dc2c0a20fe3228ddc4144854/aten/src/ATen/native/TensorCompare.cpp#L285 . This will catch it early.

matthewhagraphcore committed:
Add dtype checks in meta dispatch for argmin/argmax

matthewhagraphcore committed:
Merge branch 'main' into argminmax_dtypes

janeyx99 commented:
Test case?

hameerabbasi commented:
Thanks for the PR -- I noticed that `topk`, `sort`, `minimum`, `maximum` all have this behavior. Perhaps it makes sense to extend this PR to those functions as well?

matthewhagraphcore committed:
Merge branch 'main' into argminmax_dtypes

matthewhagraphcore committed:
Merge branch 'main' into argminmax_dtypes

matthewhagraphcore committed:
Add other ops and fix existing check

matthewhagraphcore commented:
I've added the other functions. Sort had a `TORCH_CHECK_VALUE` instead of a `TORCH_CHECK` , so to make consistent with the others I've updated that, otherwise it raised a ValueError rather than a RuntimeError, 

Not sure what that build error is about?

janeyx99 commented:
CI failures are related

hameerabbasi commented:
It seems some of the operations are implemented for bool but not complex -- we should not error on bool for those.

matthewhagraphcore committed:
Merge branch 'main' into argminmax_dtypes

matthewhagraphcore committed:
Remove minimum and maximum from bool check. Update tests

janeyx99 reviewed (approved):
thank u!

matthewhagraphcore committed:
Update sort test
